pysite/route_manager.py

```diff
@@ -3,42 +3,48 @@
 import inspect
 import os
 
-from flask import Blueprint, Flask, abort, g
-
-import rethinkdb
+from flask import Blueprint, Flask, g
 
 from pysite.base_route import APIView, BaseView, ErrorView, RouteView
-
-DB_HOST = os.environ.get("RETHINKDB_HOST")
-DB_PORT = os.environ.get("RETHINKDB_PORT")
-DB_DATABASE = os.environ.get("RETHINKDB_DATABASE")
-DB_TABLE = os.environ.get("RETHINKDB_TABLE")
+from pysite.database import RethinkDB
 
 TEMPLATES_PATH = "../templates"
 STATIC_PATH = "../static"
 
 
 class RouteManager:
     def __init__(self):
+
+        # Set up the app and the database
         self.app = Flask(
             __name__, template_folder=TEMPLATES_PATH, static_folder=STATIC_PATH, static_url_path="/static",
         )
+        self.db = RethinkDB()
         self.app.secret_key = os.environ.get("WEBPAGE_SECRET_KEY")
         self.app.config["SERVER_NAME"] = os.environ.get("SERVER_NAME", "localhost")
+        self.app.before_request(self.db.before_request)
+        self.app.teardown_request(self.db.teardown_request)
 
-        self.main_blueprint = Blueprint("main", __name__)
+        # Store the database in the Flask global context
+        g.db = self.db
```

Comment:
If you could type-hint this with a comment, our IDEs would be thankful.

`# type: RethinkDB`